### PR TITLE
Added explicit exception when Null is provided as a value for routing…

### PR DESCRIPTION
**Pull request by starkmsu**
…Key in BasicPublish

## Proposed Changes

When Null is provided as a value for routingKey in BasicPublish, some vague exception is thrown down the callstack.
Explicit ArgumentNullException is added to make things clear for such cases.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

**Comment by bording** [comment]
@starkmsu You should use `nameof(routingKey)` instead hardcoding the parameter name as a string.

**Comment by kjnilsson** [comment]
Did you try to compile this code?

C# has no default implicit conversion of `string` to `bool` so this results in the following compiler error:

```src/client/impl/ModelBase.cs(1283,17): error CS0029: Cannot implicitly convert type 'string' to 'bool' [/tmp/build/b95f7b06/rabbitmq_dotnet_client/projects/client/RabbitMQ.Client/RabbitMQ.Client.csproj]```

File: projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringModel.cs [hunk 1]
```diff
@@ -575,6 +575,11 @@ public void _Private_BasicGet(string queue, bool autoAck)
             IBasicProperties basicProperties,
             byte[] body)
         {
+            if (routingKey)
```
Comment (michaelklishin) [inline projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringModel.cs]: This doesn’t see right to me. Can we compare with `null` the old fashioned way?

**Comment by starkmsu** [comment]
Sorry, fixed this issue

**Pull request merged**